incompatibility/incompatibility-report/node-search.go

```diff
@@ -0,0 +1,64 @@
+package incompatibility
+
+import (
+	"errors"
+	"strconv"
+
+	"gopkg.in/yaml.v3"
+)
+
+type keyValue struct {
+	key   *yaml.Node
+	value *yaml.Node
+}
+
+// Given a path in a yaml file return the node at the end of the path
+func findNode(node *yaml.Node, path ...string) (*yaml.Node, error) {
```

Comment:
Sorry I missed this before.  I think this is a very functional bit of code that we can and should test directly.  It has a number of branches and would likely greatly benefit from direct testing. 

Maybe get Alans input on this.

Reply:
yeah I agree that there should be some specific testing to yaml.Node but I think the existence testing is a good start for this PR, I don't want to keep growing this too much. I'll try to add it in another appropriate PR